clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -1612,29 +1612,32 @@ bool
 check_duplicate_syscall_with_same_pc()
 {
     std::cerr << "Testing duplicate syscall\n";
-    // Incorrect: syscalls with the same PC.
 #if defined(X86_64) || defined(X86_32) || defined(ARM_64)
     constexpr addr_t ADDR = 0x7fcf3b9d;
+    // Skip test on x86-32 because PC discontinuity checks for 32-bit are disabled
+    // at syscall instrs.
+#    ifndef X86_32
+    // Incorrect: syscalls with the same PC.
     {
         std::vector<memref_t> memrefs = {
             gen_marker(TID_A, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, 64),
             gen_marker(TID_A, TRACE_MARKER_TYPE_PAGE_SIZE, 4096),
             gen_marker(TID_A, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
-#    if defined(X86_64) || defined(X86_32)
+#        if defined(X86_64) || defined(X86_32)
```

Comment:
A little weird to check for X86_32 inside ifndef X86_32.  But I guess if this were copied elsewhere we'd want this.  Not sure whether it should go or stay; would lean toward remove.

Reply:
Done.